src/commons/components/donut-chart/donut-chart.js

```diff
@@ -42,69 +42,62 @@
 
     var app = angular.module("TendrlModule");
 
-    app.directive("donutChart", donutChart);
+    app.component("donutChart", {
 
-    /*@ngInject*/
-    function donutChart(config) {
-
-        return  {
-                restrict: "C",
-
-                scope: {
+                bindings: {
                     chartTitle: "@?",
                     chartData: "=",
                     chartThresholds: "=?"
                 },
 
-                replace: false,
-
-                link: function(scope, element, attributes) {
-
-                    if(typeof scope.chartData !== "undefined") {
-                        var c3ChartDefaults = $().c3ChartDefaults();
-                        scope.customDonutChartConfig = c3ChartDefaults.getDefaultDonutConfig(scope.chartData["percent_used"]+"%");
-                        scope.customDonutChartConfig.size.height = 90;
-                        scope.customDonutChartConfig.size.width = 80;
-                        scope.customDonutChartConfig.donut.width = 10;
-                        scope.customDonutChartConfig.bindto = element[0].childNodes[0]
-                        scope.customDonutChartConfig.color =  {
-                            pattern: [scope.getThresholdColor(),"#D1D1D1"]
-                        };
-                        scope.customDonutChartConfig.data = {
-                            type: "donut",
-                            columns: [
-                              [ "Used", scope.used ],
-                              ["Available", scope.available ]
-                            ],
-                            groups: [
-                              ["used", "available"]
-                            ],
-                            order: null
-                        };
-                        c3.generate(scope.customDonutChartConfig);
-                    }
-
-                },
-                controller: function($scope, config) {
+                controllerAs: 'vm',
+                
+                controller: function($element, config) {
```

Comment:
@kamleshverma1 I will recommend passing a name of controller here and writing the function differently.
Eg: 
controller: donutChartController,

function donutChartController($element, config){}